pallets/inflation-manager/src/lib.rs

```diff
@@ -135,30 +140,16 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
-			let inflation_configuration = T::DefaultInflationConfiguration::get();
-			// install inflation config
-			InflationConfiguration::<T>::put(inflation_configuration.clone());
-
-			// set current year to 1
-			CurrentYear::<T>::put(1);
-
-			// calc inflation for first year
-			let inflation_parameters =
-				Pallet::<T>::update_inflation_parameters(&inflation_configuration);
-
-			// install inflation parameters for first year
-			InflationParameters::<T>::put(inflation_parameters.clone());
-
-			// set the flag to calculate inflation parameters after a year(in blocks)
-			let racalculation_target_block = frame_system::Pallet::<T>::current_block_number() +
-				T::BlockNumber::from(BLOCKS_PER_YEAR);
-
-			// Update recalculation flag
-			DoRecalculationAt::<T>::put(racalculation_target_block);
-
-			let block_rewards = Pallet::<T>::rewards_per_block(&inflation_parameters);
-
-			BlockRewards::<T>::put(block_rewards);
+			let do_recalculation_at = T::DoInitializeAt::get();
+
+			// if DoRecalculationAt was provided as zero,
+			// Then do TGE now and initialize inflation
+			if do_recalculation_at == T::BlockNumber::default() {
```

Comment:
Is that possible to specify it explicitly? `T::BlockNumber::from(0u32)`

Reply:
Implemented this 💯